README.md

```diff
@@ -48,47 +82,121 @@ and reference it in settings.py:
+    'AUDIENCE': 'https://api.hel.fi/auth/projects',
+    # Who we trust to sign the tokens. The library will request the
+    # public signature keys from standard locations below this URL
+    'ISSUER': 'https://api.hel.fi/sso'
+    # The following can be used if you need certain OAuth2 scopes
+    # for any functionality of the API. The request will be denied
+    # if scopes starting with API_SCOPE_PREFIX are not present
+    # in the token claims. Usually this is not needed, as checking
+    # the audience is enough.
+     REQUIRE_API_SCOPE_FOR_AUTHENTICATION': True,
+    'API_SCOPE_PREFIX': 'projects',
+}
+```
+
+### API authentication using JWT in any setup
+
+If you want to authorize access to your API using JWTs, but you are not using Django REST Framework, then this might be useful to you.
+
+API token authentication is a stateless authentication method, where every request is authenticated by checking the signature of the included JWT token.
+It still creates a persistent Django user, which is updated with the information from the token with every request.
+
+Django-helusers contains a `helusers.oidc.RequestJWTAuthentication` class.
+It has a method called `authenticate` that takes a [Django HttpRequest](https://docs.djangoproject.com/en/3.1/ref/request-response/#django.http.HttpRequest) as an argument, looks for a JWT from that request and performs authentication.
+User of this class can use it in any way they need to perform authentication and/or authorization.
+Check the class documentation for more details.
+
+Some settings are needed (and some are optional) that affect how the `RequestJWTAuthentication` class works.
+
+```python
+OIDC_API_TOKEN_AUTH = {
+    # Audience that must be present in the token for the request to be
+    # accepted. Value must be agreed between your SSO service and your
+    # application instance. Essentially this allows your application to
+    # know that the token is meant to be used with it.
+    # This setting is required.
+    'AUDIENCE': 'https://api.hel.fi/auth/projects',
+
+    # Who we trust to sign the tokens. The library will request the
+    # public signature keys from standard locations below this URL.
+    # RequestJWTAuthentication supports multiple issuers, so this
+    # setting can also be a list of strings.
+    # Default is https://tunnistamo.hel.fi.
+    'ISSUER': 'https://api.hel.fi/sso'
+
+    # The following can be used if you need certain scopes for any
+    # functionality of the API. Usually this is not needed, as checking
+    # the audience is enough. Default is False.
+    'REQUIRE_API_SCOPE_FOR_AUTHENTICATION': True,
+    # The name of the claim that is used to read in the scopes from the JWT.
+    # Default is https://api.hel.fi/auth.
+    'API_AUTHORIZATION_FIELD': 'scope_field',
+    # The request will be denied if scopes don't contain anything starting
+    # with the value provided here.
+    'API_SCOPE_PREFIX': 'projects',
```

Comment:
This works together with `REQUIRE_API_SCOPE_FOR_AUTHENTICATION`. Maybe have a shared comment for them and have the setting lines together as well.

Reply:
I grouped these three settings together to indicate that they belong together. They are separated from the rest by blank lines. I can improve the formatting somehow if that isn't clear enough.